### PR TITLE
Cleaned up ripples due to MutJS to JS type change

### LINKED ISSUE #14649: Bluetooth maps don't need to store mutable JS values
The `device_instace_map` in `Bluetooth` stores `MutJS<BluetoothDevice>` values, but it looks like they don't need to be mutable. It can use `JS<BluetoothDevice>` instead. Similarly, `BluetoothDevice`, `BluetoothRemoteGATTDescriptor`, `BluetoothRemoteGATTCharacteristic`, `BluetoothRemoteGATTServer`, and `BluetoothRemoteGATTService`.

Code: `components/script/dom/bluetooth.rs`, `components/script/dom/bluetoothdevice.rs`, `components/script/dom/bluetoothremotegattdescriptor.rs`, `components/script/dom/bluetoothremotegattcharacteristic.rs`, `components/script/dom/bluetoothremotegattserver.rs`, `components/script/dom/bluetoothremotegattservice.rs`

No need for tests; if it compiles, it's good enough for a PR.

### PR DESCRIPTION
**Pull request by 0xRampey**
Changed all Mutable JS values to JS values in the following files and fixed any differences that might have been caused due to type change.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #14649  (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because minimal type changes made.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14891)

### CODE REVIEW

**Comment by jdm** [comment]
@bors-servo: r+
Thanks for doing this!

File: components/script/dom/bluetooth.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use dom::bindings::codegen::Bindings::EventHandlerBinding::EventHandlerNonNull;
 use dom::bindings::codegen::UnionTypes::StringOrUnsignedLong;
 use dom::bindings::error::Error::{self, Network, NotFound, Security, Type};
 use dom::bindings::error::Fallible;
-use dom::bindings::js::{MutJS, Root};
+use dom::bindings::js::{ JS, Root};
```
Comment (emilio) [inline components/script/dom/bluetooth.rs]: Does this pass tidy? If so, we should make a lint to avoid it.

**Comment by emilio** [comment]
@bors-servo r-

Please remove the whitespace between `{` and `JS` in both occurrences.

File: components/script/dom/bluetooth.rs (hunk 1 above)
Comment (prampey) [inline components/script/dom/bluetooth.rs]: Yes it does.

**Comment by prampey** [comment]
Okay. Made the changes.

**Comment by emilio** [comment]
Nice, thanks @prampey!

@bors-servo r=jdm,emilio

File: components/script/dom/bluetooth.rs (hunk 1 above)
Comment (jdm) [inline components/script/dom/bluetooth.rs]: Filed #14898.

**Comment by prampey** [comment]
That's weird.  `./mach test-stylo` tests OK for me.

**Comment by emilio** [comment]
@prampey no worries, that's known and won't block your PR from merging :)

**Comment by prampey** [comment]
Oh okay. Thanks :)

**Comment by jdm** [comment]
It says linking failed with no other output. I suspect the builder is running out of disk space or something like that, but let's see if it keeps happening:
@bors-servo: retry

**Pull request merged**